zetaclient/tss_signer.go

```diff
@@ -334,23 +339,50 @@ func getKeyAddrBTCWitnessPubkeyHash(tssPubkey string) (*btcutil.AddressWitnessPu
 	return addr, nil
 }
 
-func NewTSS(peer p2p.AddrList, privkey tmcrypto.PrivKey, preParams *keygen.LocalPreParams, cfg *config.Config, bridge *ZetaCoreBridge) (*TSS, error) {
+func NewTSS(peer p2p.AddrList, privkey tmcrypto.PrivKey, preParams *keygen.LocalPreParams, cfg *config.Config, bridge *ZetaCoreBridge, tssHistoricalList []types.TSS) (*TSS, error) {
 	server, err := SetupTSSServer(peer, privkey, preParams, cfg)
 	if err != nil {
 		return nil, fmt.Errorf("SetupTSSServer error: %w", err)
 	}
 	tss := TSS{
-		Server:     server,
-		Keys:       make(map[string]*TSSKey),
-		logger:     log.With().Str("module", "tss_signer").Logger(),
-		coreBridge: bridge,
+		Server:        server,
+		Keys:          make(map[string]*TSSKey),
+		CurrentPubkey: cfg.CurrentTssPubkey,
+		logger:        log.With().Str("module", "tss_signer").Logger(),
+		coreBridge:    bridge,
 	}
 
-	files, err := os.ReadDir(cfg.TssPath)
+	err = tss.LoadTssFilesFromDirectory(cfg.TssPath)
+	if err != nil {
+		return nil, err
+	}
+	err = tss.VerifyKeysharesForPubkeys(tssHistoricalList)
 	if err != nil {
-		fmt.Println("ReadDir error", err)
 		return nil, err
 	}
+	return &tss, nil
+}
+
+func (tss *TSS) VerifyKeysharesForPubkeys(tssList []types.TSS) error {
```

Comment:
i don't think this is correct, new keysigners would not have old keyshares that they were not part of.

Reply:
Thanks that's a great catch , 
Modified to using only TSS that the node participated in